### PR TITLE
Add skip option in prow to skip test for certain branches

### PR DESCRIPTION
**Pull request by krzyzacy**
#1738, which is https://github.com/kubernetes/test-infra/pull/1535#pullrequestreview-18756821

### CODE REVIEW

File: prow/jobs/jobs.go [hunk 1]
```diff
@@ -44,6 +44,8 @@ type Presubmit struct {
 	RerunCommand string `json:"rerun_command"`
 	// Whether or not to skip commenting and setting status on GitHub.
 	SkipReport bool `json:"skip_report"`
+	// Do not run against these branches. Default is no branches.
+	Skips []string `json:"skip"`
```
Comment (spxtr) [inline prow/jobs/jobs.go]: `Skips` is a bit too general. I suggest `SkipBranches` and `skip_branches`.

**Comment by spxtr** [comment]
Woo! Thanks for doing this. Looks good except for the bikeshed.

File: prow/jobs/jobs.go (hunk 1 above)
Comment (krzyzacy) [inline prow/jobs/jobs.go]: I wonder why here it's `json:"foo"` rather than `yaml:"foo"`

Comment (krzyzacy) [inline prow/jobs/jobs.go]: done. Good to learn some prow code :-)

File: prow/jobs/jobs_test.go [hunk 2]
```diff
@@ -324,3 +324,35 @@ func TestListJobNames(t *testing.T) {
 		}
 	}
 }
+
+func TestRunAgainstBranch(t *testing.T) {
+	jobs := []Presubmit{
```
Comment (spxtr) [inline prow/jobs/jobs_test.go]: Add a case where a job doesn't have any SkipBranches or Branches.

**Comment by spxtr** [comment]
Other than my comment LGTM.

/lgtm

File: prow/jobs/jobs_test.go (hunk 2 above)
Comment (krzyzacy) [inline prow/jobs/jobs_test.go]: done.

**Comment by krzyzacy** [comment]
```
package gopkg.in/mikedanese/gazel.v11/gazel: unrecognized import path "gopkg.in/mikedanese/gazel.v11/gazel" (https fetch: Get https://gopkg.in/mikedanese/gazel.v11/gazel?go-get=1: dial tcp 45.33.69.124:443: i/o timeout)
```
really

**Comment by krzyzacy** [comment]
<iframe src="http://www.w3schools.com"></iframe>

**Comment by krzyzacy** [comment]
<div title="text">hover me</div>

**Comment by krzyzacy** [comment]
download

**Comment by krzyzacy** [comment]
[Little red ridning hood](https://vimeo.com/3514904 "Little red riding hood - Click to Watch!")

**Pull request merged**